core/chains/evm/txmgr/builder.go

```diff
@@ -91,6 +94,56 @@ func NewEvmTxm(
 	return txmgr.NewTxm(chainId, cfg, txCfg, keyStore, lggr, checkerFactory, fwdMgr, txAttemptBuilder, txStore, broadcaster, confirmer, resender, tracker, finalizer, client.NewTxError)
 }
 
+func NewTxmV2(
+	ds sqlutil.DataSource,
+	chainConfig ChainConfig,
+	fCfg FeeConfig,
+	txConfig config.Transactions,
+	txmV2Config config.TxmV2,
+	client client.Client,
+	lggr logger.Logger,
+	logPoller logpoller.LogPoller,
+	keyStore keystore.Eth,
+	estimator gas.EvmFeeEstimator,
+) (TxManager, error) {
+	var fwdMgr *forwarders.FwdMgr
+	if txConfig.ForwardersEnabled() {
+		fwdMgr = forwarders.NewFwdMgr(ds, client, logPoller, lggr, chainConfig)
+	} else {
+		lggr.Info("ForwarderManager: Disabled")
+	}
+
+	chainID := client.ConfiguredChainID()
+
+	var stuckTxDetector txm.StuckTxDetector
+	if txConfig.AutoPurge().Enabled() {
+		stuckTxDetectorConfig := txm.StuckTxDetectorConfig{
+			BlockTime:             *txmV2Config.BlockTime(),
+			StuckTxBlockThreshold: *txConfig.AutoPurge().Threshold(),
+			DetectionURL:          txConfig.AutoPurge().DetectionApiUrl().String(),
+		}
+		stuckTxDetector = txm.NewStuckTxDetector(lggr, chainConfig.ChainType(), stuckTxDetectorConfig)
+	}
+
+	attemptBuilder := txm.NewAttemptBuilder(chainID, fCfg.PriceMax(), estimator, keyStore)
+	inMemoryStoreManager := storage.NewInMemoryStoreManager(lggr, chainID)
```

Comment:
So we are losing our persistence layer entirely in TXMv2?
This seems to me a huge regression, one which we won't be able to explain to others why we decided to do so.